main.c

```diff
@@ -280,91 +383,17 @@ main(int argc, char *argv[])
 		/* User selected 'info' mode. */
 		if (scrypt_mode_info(infilename))
 			goto err0;
-		rc = SCRYPT_OK;
-		goto done;
-	}
-
-	/* If the input isn't stdin, open the file. */
```

Comment:
Why not keep this here?  I don't see any benefit to duplicating this block between the two scrypt_mode functions.